### ISSUE TITLE
Solspace\Freeform\Services\MailerService::processRecipients(): Argument #1 ($recipients) must be of type Solspace\Freeform\Notifications\Components\Recipients\RecipientCollection, array given

### ISSUE THREAD

**Issue by jcdarwin** [solspace/craft3-freeform#1253]
**Describe the bug or issue you're experiencing**

I've set up an export notification that has two email addresses entered in the `Emails to Notify` field, each on their own line.

However, when the export attempts to run, we receive the following error in Sentry (and no export is produced):

```
TypeError
Solspace\Freeform\Services\MailerService::processRecipients(): Argument #1 ($recipients) must be of type Solspace\Freeform\Notifications\Components\Recipients\RecipientCollection, array given, called in /var/www/chinacapable/releases/20240426041212/chinacapable/craft/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Notifications/Export/ExportNotifications.php on line 88
```

**Steps to reproduce**

1. Set up an export notification that has two email addresses entered in the `Emails to Notify` field, each on their own line.

2. Wait for the export to process -- you should see the error mentioned above  in the logs

**Expected behavior**

The export file should be produced successfully.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.8.10
- Freeform Version: 5.1.15
- Freeform Edition: Pro
- Fresh Install or Upgrade: Fresh Install

**Additional context**

Looking at the code, `processRecipients`  is indeed typed to accept `$recipients` as an `RecipientCollection`:

```
    public function processRecipients(RecipientCollection $recipients): array
    {
        if (version_compare(\Craft::$app->getVersion(), '3.5', '>=')) {
            $testToEmailAddress = \Craft::$app->getConfig()->getGeneral()->getTestToEmailAddress();
            if (!empty($testToEmailAddress)) {
                return $testToEmailAddress;
            }
        }

        return $recipients->emailsToArray();
    }
```

However, where it is called (`craft/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Notifications/Export/ExportNotifications.php`) it simply decodes the string:

```
$message->setTo($mailer->processRecipients(json_decode($notification->recipients)));
```

Presumably this should instead be something like:

```
$recipients = new RecipientCollection(json_decode($notification->recipients));
$message->setTo($mailer->processRecipients($recipients));
```

**Comment by gustavs-gutmanis** [comment]
Thanks for reporting this @jcdarwin, this has slipped through the cracks during the FF5 migration.

We will fix this and release it in the upcoming version!

**Comment by kjmartens** [comment]
This is available now in Freeform 5.1.17+. 🙂
